tutorials/tts/4_TTS_FastPitch_Finetuning.ipynb

```diff
@@ -0,0 +1,972 @@
+       "          (0): Conv1d(1, 128, kernel_size=(15,), stride=(1,), padding=(7,))\n",
+       "          (1): Conv1d(128, 128, kernel_size=(41,), stride=(2,), padding=(20,), groups=4)\n",
+       "          (2): Conv1d(128, 256, kernel_size=(41,), stride=(2,), padding=(20,), groups=16)\n",
+       "          (3): Conv1d(256, 512, kernel_size=(41,), stride=(4,), padding=(20,), groups=16)\n",
+       "          (4): Conv1d(512, 1024, kernel_size=(41,), stride=(4,), padding=(20,), groups=16)\n",
+       "          (5): Conv1d(1024, 1024, kernel_size=(41,), stride=(1,), padding=(20,), groups=16)\n",
+       "          (6): Conv1d(1024, 1024, kernel_size=(5,), stride=(1,), padding=(2,))\n",
+       "        )\n",
+       "        (conv_post): Conv1d(1024, 1, kernel_size=(3,), stride=(1,), padding=(1,))\n",
+       "      )\n",
+       "    )\n",
+       "    (meanpools): ModuleList(\n",
+       "      (0): AvgPool1d(kernel_size=(4,), stride=(2,), padding=(2,))\n",
+       "      (1): AvgPool1d(kernel_size=(4,), stride=(2,), padding=(2,))\n",
+       "    )\n",
+       "  )\n",
+       "  (feature_loss): FeatureMatchingLoss()\n",
+       "  (discriminator_loss): DiscriminatorLoss()\n",
+       "  (generator_loss): GeneratorLoss()\n",
+       ")"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from nemo.collections.tts.models import HifiGanModel\n",
+    "from nemo.collections.tts.models import FastPitchModel\n",
+    "\n",
+    "hifigan_ckpt_path = \"/home/pneekhara/PreTrainedModels/HifiGan--val_loss=0.08-epoch=899.ckpt\"\n",
+    "vocoder = HifiGanModel.load_from_checkpoint(hifigan_ckpt_path)\n",
+    "vocoder.eval().cuda()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "0a4c986f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def infer(spec_gen_model, vocoder_model, str_input, speaker = None):\n",
+    "    parser_model = spec_gen_model\n",
+    "    with torch.no_grad():\n",
+    "        parsed = parser_model.parse(str_input)\n",
+    "        if speaker is not None:\n",
+    "            speaker = torch.tensor([speaker]).long().cuda()\n",
+    "        spectrogram = spec_gen_model.generate_spectrogram(tokens=parsed, speaker = speaker)\n",
+    "        audio = vocoder_model.convert_spectrogram_to_audio(spec=spectrogram)\n",
+    "        \n",
+    "    if spectrogram is not None:\n",
+    "        if isinstance(spectrogram, torch.Tensor):\n",
+    "            spectrogram = spectrogram.to('cpu').numpy()\n",
+    "        if len(spectrogram.shape) == 3:\n",
+    "            spectrogram = spectrogram[0]\n",
+    "    if isinstance(audio, torch.Tensor):\n",
+    "        audio = audio.to('cpu').numpy()\n",
+    "    return spectrogram, audio\n",
+    "\n",
+    "def get_best_ckpt(experiment_base_dir, new_speaker_id, duration_mins, mixing_enabled, original_speaker_id):\n",
```

Comment:
Please add a short docstring to explain what this function does.

Reply:
added